extractor_zos.go

```diff
@@ -0,0 +1,35 @@
+// +build zos
+package fastzip
+
+import (
+    "os"
+    "time"
+)
+
+func lchmod(name string, mode os.FileMode) error {
+    if mode&os.ModeSymlink != 0 {
+        // it is a symlink, do not follow
+        return nil
+    }
+    err := os.Chmod(name, mode)
+    if err != nil {
+        return &os.PathError{Op: "lchmod", Path: name, Err: err}
+    }
+    return nil
+}
+
+func lchtimes(name string, mode os.FileMode, atime, mtime time.Time) error {
+    err := lchmod(name, mode)
+    if err != nil {
+        return err
+    }
+    err = os.Chtimes(name, atime, mtime)
+    if err != nil {
+        return &os.PathError{Op: "lchtimes", Path: name, Err: err}
+    }
+    return nil
```

Comment:
I'm assuming that `lutimes` is supported by `zos`, but it's available as part of `golang.org/x/sys/unix` yet?

I wonder if we can support it by calling the syscall directly. Are there any plans for `zos` support to be added to `golang.org/x/sys/unix`?

Reply:
I need to check internally.

Reply:
`lutimes` is not yet available natively on zos < 2.5. The implementation here is a workaround. We have tested this internally with fastzip APIs.